volume/dir.go

```diff
@@ -7,11 +7,18 @@ import (
 	"github.com/puppetlabs/wash/plugin"
 )
 
+// keystone represents an entry that can be used as a key, as well as the volume path associated
+// with that key. We must always query the same path when running VolumeList for its related key.
```

Comment:
Still not quite sure I understand the issue. Some examples of where things could go wrong would be useful.

Reply:
I'll work on a better comment with an example.

Reply:
I wrote up what I think would happen. I haven't found a combination of commands that actually make it happen, but I think my reasoning is sound.

Reply:
Oh I think I see the issue now (the comment helped). Entries rooted at the same node (e.g. `/`, or `/1/2/.../<maxdepth>` or `/<maxdepth+1>/.../<maxdepth+maxdepth>`) should enumerate the root whenever the cache runs out but instead they enumerate themselves.

I think this would be clearer if we changed `key plugin.Entry` => `root *dir` (or a better name than `root`, maybe `subtreeRoot` or `enumeratingParent`?). That way, `volume.List` takes in `root.path` instead of `key.string` and it's clear that the `root` is the cache key b/c it is where we're enumerating from.

Reply:
Oh right, don't need a new type for it.

Reply:
It's not quite that simple. The root we start from isn't a `*dir`, so I need to special-case to use impl for that, and it gets kind of messy.

Reply:
Actually I don't see a way to clean that up without doing what I have here. I need to attach a path to the root entry, which means making something that wraps a plugin.Entry. But we don't want to implement `dir` that way because we need to set things on an EntryBase for it.

Reply:
Updated.